projects/netgrif-components-core/src/lib/task-content/services/task-content.service.ts

```diff
@@ -314,6 +334,24 @@ export abstract class TaskContentService implements OnDestroy {
     private isFieldInTaskRef(changedField: string): boolean {
         return !!this.taskFieldsIndex &&
             Object.keys(this.taskFieldsIndex)
-            .some(taskId => taskId !== this.task.stringId && this.taskFieldsIndex[taskId][changedField]);
+            .some(taskId => taskId !== this.task.stringId && this.taskFieldsIndex[taskId].fields[changedField]);
+    }
+
+    private getReferencedTransitionId(changedField: string): string {
+        let taskFieldsIndexId = '';
+        if (!!this.taskFieldsIndex) {
+            taskFieldsIndexId = Object.keys(this.taskFieldsIndex).find(taskId =>
+                Object.keys(this.taskFieldsIndex[taskId].fields).includes(changedField));
+        }
+        return this.taskFieldsIndex[taskFieldsIndexId].transitionId;
```

Comment:
if `this.taskFieldsIndex` is falsty it still gets accessed with the `[]` operator and will almost certainly throw an error.

Reply:
Corrected